### PR TITLE
[MCKIN-24505] [BB-2953] Python 3 Support

### PR DESCRIPTION
**Pull request by shimulch**
This PR adds python 3 support for this xblock. This is required by edX Juniper release.

**JIRA tickets**: 
- https://tasks.opencraft.com/browse/BB-2953

~~**Discussions**:~~

**Dependencies**: New dependencies -
- ``future==0.18.2``

**Screenshots**: N/A

~~**Sandbox URL**:~~

**Merge deadline**: None

**Testing instructions**:

1. This xblock should work on python 3

**Author notes and concerns**: TBD

**Reviewers**
- [ ] @xitij2000 

~~**Settings**~~

### CODE REVIEW

File: run_tests.py [hunk 1]
```diff
@@ -20,11 +22,13 @@
     # Use the workbench settings file:
     os.environ.setdefault("DJANGO_SETTINGS_MODULE", "workbench.settings")
     # Configure a range of ports in case the default port of 8081 is in use
-    os.environ.setdefault("DJANGO_LIVE_TEST_SERVER_ADDRESS", "localhost:8081-8099")
+    # os.environ.setdefault("DJANGO_LIVE_TEST_SERVER_ADDRESS", "localhost:8081-8099")
```
Comment (xitij2000) [inline run_tests.py:25]:
```suggestion

```

File: run_tests.py [hunk 2]
```diff
@@ -20,11 +22,13 @@
...
 
     # Silence too verbose Django logging
     logging.disable(logging.DEBUG)
 
+    settings.INSTALLED_APPS.append('problem_builder')
```
Comment (xitij2000) [inline run_tests.py:30]: Instead of doing this, please create a new settings file. That way it will be possible to run other commands with that context.

File: tests/integration/test_eoc_journal.py [hunk 3]
```diff
@@ -485,7 +487,7 @@ def test_engagement_metrics_listed_in_student_view_when_enabled(self):
 
         element = self.get_element_for_course_engagement()
 
-        for key, value in default_engagement_metrics.iteritems():
+        for key, value in list(default_engagement_metrics.items()):
```
Comment (xitij2000) [inline tests/integration/test_eoc_journal.py:490]: I don't think a list conversion is needed here. 
```suggestion
        for key, value in default_engagement_metrics.items():
```

File: tests/integration/test_eoc_journal.py [hunk 4]
```diff
@@ -224,7 +226,7 @@ def get_element_for_course_engagement(self):
 
     def select_option_in_boolean_field(self, field_name, value=True):
         control = self.get_element_for_field(field_name)
-        select = Select(control.find_element_by_id("xb-field-edit-{}".format(field_name)))
+        select = Select(control)
```
Comment (xitij2000) [inline tests/integration/test_eoc_journal.py:229]: Why is this change needed?

File: manage.py [hunk 5]
```diff
@@ -1,10 +1,15 @@
...
+from django.conf import settings
 
+if __name__ == "__main__":
+    xblock_sdk_dir = os.path.dirname(os.path.dirname(workbench.__file__))
+    sys.path.append(xblock_sdk_dir)
+    os.environ.setdefault("DJANGO_SETTINGS_MODULE", "workbench.settings")
+    settings.INSTALLED_APPS.append('problem_builder')
     from django.core.management import execute_from_command_line
 
     execute_from_command_line(sys.argv)
```
Comment (xitij2000) [inline manage.py:15]: Don't modify this, this is for building the translations, so it needs different settings.

File: eoc_journal/eoc_journal.py [hunk 6]
```diff
@@ -337,10 +341,9 @@ def list_user_pb_answers_by_section(self):
             # Make list of sections-answers
             return [
                 {'name': key, 'questions': value}
-                for key, value in answers.items()
+                for key, value in list(answers.items())
```
Comment (xitij2000) [inline eoc_journal/eoc_journal.py:344]:
```suggestion
                for key, value in answers.items()
```

**Review by xitij2000 — CHANGES_REQUESTED** [review]
Please switch all error codes to use names like "import-outside-toplevel" instead of "C0415".

File: tests/integration/test_eoc_journal.py (hunk 4 above)
Comment (shimulch) [reply tests/integration/test_eoc_journal.py:229]: Somehow ``control`` itself is the ``xb-field-edit-{}`` control, which previously was the control wrapper (``li``) element. This might be due to changes in ``xblock-sdk`` version. But I didn't quite understand why two versions of python getting two different behaviour for ``get_element_for_field`` method.

Comment (xitij2000) [reply tests/integration/test_eoc_journal.py:229]: It's likely because of a change in some library. Might be worth investigating.

**Review by xitij2000 — CHANGES_REQUESTED** [review]
Could you look into and clean up some of the dependencies if you have time? If not it's OK.

**Review by xitij2000 — APPROVED** [review]
This is good to go but there is an issue with auth that needs to be fixed before this can be tested properly

File: circle.yml [hunk 7]
```diff
@@ -22,9 +23,64 @@ jobs:
...
+          command: |
+            virtualenv venv
+            . venv/bin/activate
+            pip install -U pip wheel
+            pip install -r requirements-test.txt
+            pip install -r venv/src/xblock-sdk/requirements/base.txt
+            pip install -r venv/src/xblock-sdk/requirements/test.txt
+            mkdir var
+      - save_cache:
+          key: dependencies-{{ checksum "circle.yml" }}-{{ checksum "requirements.txt" }}-{{ checksum "requirements-test.txt" }}
```
[the remark this answers is not in the document]
Comment (shimulch) [reply circle.yml]: Updated. Both keys are now the same.

**Comment by xitij2000** [comment]
@shimulch I've added a commit to fix an issue I had when running this on solutions. It seems that the course id is a bytestring which causes issues. So I've coerced it to unicode.

**Comment by shimulch** [comment]
@xitij2000 
LGTM :+1:

- [x] I read through the code, checked if str in python future 
- [N/A] I checked for accessibility issues
- [N/A] Includes documentation
- [N/A] I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.
